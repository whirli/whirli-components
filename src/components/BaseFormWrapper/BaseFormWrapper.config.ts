// @ts-ignore
import { PropType } from '@composition';
import * as PropHelpers from '@whirli-components/helpers/props';
import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseFormWrapper/BaseFormWrapper.config';

// Package config
import { PropKeys, PropValues } from './BaseFormWrapper.constants';
import type { PropMethod, PropEncode } from './BaseFormWrapper.types';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigAttributes: Record<string, Record<string, string>> = {
  [PropKeys.ENCODE]: {
    default: 'application/x-www-form-urlencoded',
    media: 'multipart/form-data',
  },
};

export const ConfigProps: ComponentProps = {
  [PropKeys.ACTION]: {
    required: true,
    type: String as PropType<string>,
  },
  [PropKeys.METHOD]: {
    required: false,
    type: String as PropType<PropMethod>,
    default: 'post',
  },
  [PropKeys.ENCODE]: {
    required: false,
    type: String as PropType<PropEncode>,
    default: 'default',
  },
  ...config.props,
};
