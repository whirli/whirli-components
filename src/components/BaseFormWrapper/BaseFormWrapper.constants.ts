export const PropKeys: Record<string, string> = {
  ACTION: 'action',
  METHOD: 'method',
  ENCODE: 'encode',
};

export const PropValues: Record<string, string[]> = {
  [PropKeys.METHOD]: ['get', 'post'],
  [PropKeys.ENCODE]: ['default', 'media'],
};
