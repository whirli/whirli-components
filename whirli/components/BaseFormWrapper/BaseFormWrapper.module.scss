@import '@whirli-components/components/BaseFormWrapper/BaseFormWrapper.module.scss';

// Standard styles

// Breakpoint styles
$localBreakpointStyles: ();

@include createBreakpointClasses(map-merge($packageBreakpointStyles, $localBreakpointStyles));