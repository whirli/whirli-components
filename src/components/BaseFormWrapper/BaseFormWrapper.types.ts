import { PropKeys, PropValues } from './BaseFormWrapper.constants';

const propMethod: string[] = PropValues[PropKeys.METHOD];
const propEncode: string[] = PropValues[PropKeys.ENCODE];

export type PropMethod = typeof propMethod[number];
export type PropEncode = typeof propEncode[number];
