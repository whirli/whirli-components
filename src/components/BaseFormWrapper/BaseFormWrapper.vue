<template>
  <form :action="props.action" :method="props.method" :enctype="attributes[PropKeys.ENCODE]">
    <slot />
  </form>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormWrapper/BaseFormWrapper.module.scss?module';

// Data
import { ConfigStyles, ConfigProps, ConfigAttributes } from './BaseFormWrapper.config';

// Constants
import { PropKeys } from './BaseFormWrapper.constants';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;
const ComponentAttributes: Record<string, Record<string, string>> = ConfigAttributes;

// @ts-ignore
const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes = makeClasses(ComponentStyles, props, styles);

// Attributes
const attributes: Record<string, string> = {
  [PropKeys.ENCODE]: ComponentAttributes[PropKeys.ENCODE][props[PropKeys.ENCODE]],
};
</script>
