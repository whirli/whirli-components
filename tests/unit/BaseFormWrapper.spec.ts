import { shallowMount } from '@vue/test-utils';
import BaseFormWrapper from '@/components/BaseFormWrapper/BaseFormWrapper.vue';

let wrapper: any;
const text = 'Hello World';

describe('BaseFormWrapper.vue - Single classes', () => {
  beforeEach(() => {
    wrapper = shallowMount(BaseFormWrapper, {
      slots: {
        default: text,
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the correct text', () => {
    expect(wrapper.text()).toMatch(text);
  });
});
