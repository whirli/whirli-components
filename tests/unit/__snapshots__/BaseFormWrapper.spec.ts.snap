// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormWrapper.vue - Single classes renders the component 1`] = `
<form
  enctype="application/x-www-form-urlencoded"
  method="post"
>
  
  Hello World
  
</form>
`;
